### ISSUE TITLE
Sik-Ho Tang | Brief Review -- Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#110]
Sik-Ho Tang. [Brief Review — Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs)](https://sh-tsang.medium.com/brief-review-fast-and-accurate-deep-network-learning-by-exponential-linear-units-elus-1daaecf1a082).

**Comment by NorbertZheng** [comment]
## Overview
Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs), ELU, by Johannes Kepler University, 2016 ICLR, Over 5000 Citations. Image Classification, Autoencoder, Activation Function, [ReLU](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf), [Leaky ReLU](https://sh-tsang.medium.com/brief-review-rectifier-nonlinearities-improve-neural-network-acoustic-models-dbc79e373854).

 - Exponential Linear Units (ELUs) are proposed as activation function.

**Comment by NorbertZheng** [comment]
## Overview
Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs), ELU, by Johannes Kepler University, 2016 ICLR, Over 5000 Citations. Image Classification, Autoencoder, Activation Function, [ReLU](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf), [Leaky ReLU](https://sh-tsang.medium.com/brief-review-rectifier-nonlinearities-improve-neural-network-acoustic-models-dbc79e373854).

 - Exponential Linear Units (ELUs) are proposed as activation function.

**Comment by NorbertZheng** [comment]
## Exponential Linear Unit (ELU)
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/61f4dd49-f9d4-47e1-8f65-9e8976b171ac)
The rectified linear unit ([ReLU](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf)), the [leaky ReLU](https://sh-tsang.medium.com/brief-review-rectifier-nonlinearities-improve-neural-network-acoustic-models-dbc79e373854) ([LReLU](https://sh-tsang.medium.com/brief-review-rectifier-nonlinearities-improve-neural-network-acoustic-models-dbc79e373854), α= 0.1), the shifted ReLUs (SReLUs), and the exponential linear unit (ELU, α = 1.0).

The ELU hyperparameter $\alpha$ controls the value to which an ELU saturates for negative net inputs:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/db43d084-9af7-4ec5-a281-086366abd934)

**Comment by NorbertZheng** [comment]
## Results
### MNIST
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ce89b180-f796-488a-a61f-84b599599caf)
(a): median of the average unit activation for different activation functions. (b): Training cross entropy loss.

 - The network had eight hidden layers of 128 units each.

**ELUs stay have smaller median throughout the training process.** The training error of ELU networks decreases much more rapidly than for the other networks.

**Comment by NorbertZheng** [comment]
### Autoencoder
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/d163e555-1a1c-44c6-8f51-3c6d45b64ae1)
Autoencoder training on MNIST: Reconstruction error for the test and training data set over epochs, using different activation functions and learning rates.

 - The encoder part consisted of four fully connected hidden layers with sizes 1000, 500, 250 and 30, respectively. The decoder part was symmetrical to the encoder.

ELUs outperform the competing activation functions in terms of training / test set reconstruction error for all learning rates.

**Comment by NorbertZheng** [comment]
### CIFAR-100
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/7f350302-9cc0-4aea-95dc-08ded1738b7b)
Comparison of [ReLU](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf)s, [LReLUs](https://sh-tsang.medium.com/brief-review-rectifier-nonlinearities-improve-neural-network-acoustic-models-dbc79e373854), and SReLUs on CIFAR-100. (a-c) show the training loss, (d-f) the test classification error.

 - The CNN for these CIFAR-100 experiments consists of 11 convolutional layers.

ELU networks achieved lowest test error and training loss.

**Comment by NorbertZheng** [comment]
### CIFAR-10 & CIFAR-100
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/cab0a524-483b-4f7a-8e3c-99188cfdd25a)
Comparison of ELU networks and other CNNs on CIFAR-10 and CIFAR-100.

 - The CNN architecture is more sophisticated than in the previous subsection and consists of 18 convolutional layers.

ELU-networks are the second best on CIFAR-10 with a test error of 6.55% but still they are among the top 10 best results reported for CIFAR-10. ELU networks performed best on CIFAR-100 with a test error of 24.28%. This is the best published result on CIFAR-100.

**Comment by NorbertZheng** [comment]
### ImageNet
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/8053cdbb-1b5d-4266-9dc9-af62fadb45ee)
ELU networks applied to ImageNet.

 - A 15 layer CNN with SPP layer, originated in [SPPNet](https://medium.com/coinmonks/review-sppnet-1st-runner-up-object-detection-2nd-runner-up-image-classification-in-ilsvrc-906da3753679?source=post_page---------------------------), is used.

The ELU-network already reaches the 20% top-5 error after 160k iterations, while the [ReLU](https://sh-tsang.medium.com/brief-review-rectified-linear-units-improve-restricted-boltzmann-machines-43724ede3ecf) network needs 200k iterations to reach the same error rate.

**Comment by NorbertZheng** [comment]
## Reference
 - [2016 ICLR] [ELU] [Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs)](https://arxiv.org/abs/1511.07289).
